pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -312,5 +315,23 @@ func (pc *PodCache) getPodByProxy(proxy *model.Proxy) *v1.Pod {
 	// because multiple ips belong to the same pod
 	proxyIP := proxy.IPAddresses[0]
 	// just in case the proxy ID is bad formatted
-	return pc.getPodByIP(proxyIP)
+	pods := pc.getPodsByIP(proxyIP)
+	switch len(pods) {
+	case 0:
+		return nil
+	case 1:
+		return pods[0]
+	default:
+		// This should only happen with hostNetwork pods, which cannot be proxy clients...
+		log.Errorf("unexpected: found multiple pods for proxy %v (%v)", proxy.ID, proxyIP)
+		// Try to handle it gracefully
+		for _, p := range pods {
+			// At least filter out wrong namespaces...
+			if proxy.ConfigNamespace != p.Namespace {
```

Comment:
nit: can abstract and resue with above https://github.com/istio/istio/pull/47893/files#diff-105059276d8259ba0abc0b02be5c15c7aefa7ab6f25624141329204ddc0f4c08R76